src/services/codefixes/importFixes.ts

```diff
@@ -485,12 +485,7 @@ namespace ts.codefix {
 
     function getFixesInfoForNonUMDImport({ sourceFile, program, cancellationToken, host, preferences }: CodeFixContextBase, symbolToken: Identifier): FixesInfo | undefined {
         const checker = program.getTypeChecker();
-        // If we're at `<Foo/>`, we must check if `Foo` is already in scope, and if so, get an import for `React` instead.
-        const symbolName = isJsxOpeningLikeElement(symbolToken.parent)
-            && symbolToken.parent.tagName === symbolToken
-            && (isIntrinsicJsxName(symbolToken.text) || checker.resolveName(symbolToken.text, symbolToken, SymbolFlags.All, /*excludeGlobals*/ false))

```

Comment:
The new code makes sense, but I don't see why the old code was broken; for the `<Text>` case, I thought that `Text` was not imported, so `checker.resolveName` should return `undefined` (and also `Text` is not an intrinsic, right?). What am I missing here?

Reply:
@sandersn Thanks for the review 👍! 

If I remember correctly I followed the following steps:
1. Need to handle names from two types of JSX tags (open and close)
1.1. Need to check if there is React if no then need to import `React` (`checker.resolveName(jsxNamespace...)`)
2. If there is React, need to return the name

The previous version just returns React even it exists and does it only for `JsxOpeningLikeElement`.  What do you think does that make sense? Or there is a better way?

Reply:
Well, the previous version only returns React if it *can* resolve `Text` via `checker.resolveName(symbolToken.text,...)`. This seems right, but clearly it produces the wrong results. I just don't understand why.

Reply:
@sandersn The problem was with `JsxOpeningLikeElement`, which was handled in a previous version :). `JsxClosingElement` worked fine, because the following condition related only to `JsxOpeningLikeElement`. As I understood this condition needs to import default `React` if a name is JSX tag. 
```ts
isJsxOpeningLikeElement(symbolToken.parent) &&
    symbolToken.parent.tagName === symbolToken && 
    (isIntrinsicJsxName(symbolToken.text) ||
     checker.resolveName(symbolToken.text, symbolToken, SymbolFlags.All, /*excludeGlobals*/ false))
```

In the new version, if a token related to JSX, and only if there is no React, we suppose a user needs to import Rect first, otherwise return name. 

```ts
<[|Text|]></[|Text|]> // QF: [there is no React] Import React
```

```ts
import React from 'react'
<[|Text|]></[|Text|]> // QF: [there is React] Import Text
```

```ts
import React from 'react'
import { Text } from 'react-native'
<Text></Text>
```

Reply:
@sandersn is it answer on your question?

Reply:
For the `JsxOpeningLikeElement`

```ts
checker.resolveName(symbolToken.text, symbolToken, SymbolFlags.All, /*excludeGlobals*/ false))
```

resolves `Text` and **returns** `jsxNamespace` not `Text`, which causes 

![1](https://user-images.githubusercontent.com/509265/83556933-6fff7700-a519-11ea-9b31-4f6998567ddf.gif)

I changed this to check if `jsxNamespace` was added, otherwise use the name

![2](https://user-images.githubusercontent.com/509265/83556975-8279b080-a519-11ea-8287-e8840556a123.gif)

Reply:
I see. I don't understand how `checker.resolveName` is able to resolve `Text` without it being imported. (That's why I thought that it would return `undefined`.)

Reply:
@sandersn I suppose `/*excludeGlobals*/ false` makes it possible.